.github/actions/integration-test/action.yml

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+name: Integration test
+description: Run the integration tests against a single version of k8s and Vault
+inputs:
+  k8s-version:
+    description: 'Kubernetes version to use for the kind cluster'
+    required: true
+  vault-version:
+    description: 'Vault version to use for the tests'
+    required: true
+  tarball-file:
+    description: 'Name of the tarball file artifact to download'
+    required: true
+  kind-cluster-name:
+    description: 'Name of the kind cluster to create and test against'
+    default: 'vault-k8s'
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+    - uses: actions/setup-node@5e21ff4d9bc1a8cf6de233a3057d20ec6b3fb69d # v3.8.1
+      with:
+        node-version: "19.8.1"
+    - run: npm install -g bats@1.10.0
+      shell: bash
+    - run: bats -v
+      shell: bash
+
+    - uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c # v5.0.0
+      with:
+        python-version: '3.10'
+    - run: pip install yq
+      shell: bash
+
+    # Checkout this repo.
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        path: "vault-k8s"
+
+    # Checkout vault-helm for acceptance test code.
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        repository: "hashicorp/vault-helm"
+        ref: "v0.27.0"
+        path: "vault-helm"
+
+    - name: Create Kind Cluster
+      uses: helm/kind-action@99576bfa6ddf9a8e612d83b513da5a75875caced # v1.9.0
+      with:
+        cluster_name: ${{ inputs.kind-cluster-name }}
+        config: vault-helm/test/kind/config.yaml
+        node_image: kindest/node:v${{ inputs.k8s-version }}
+        version: "v0.22.0"
+
+    - name: Create kind export log root
+      id: create_kind_export_log_root
+      shell: bash
+      run: |
+        vault_flavor=community
+        log_artifact_name="kind-${{ inputs.kind-cluster-name }}-$(git rev-parse --short ${{ github.sha }})-${{ inputs.k8s-version }}-${{ inputs.vault-version }}-${vault_flavor}-helm-logs"
+        log_root="/tmp/${log_artifact_name}"
+        mkdir -p "${log_root}"
+        echo "log_root=${log_root}" >> $GITHUB_OUTPUT
+        echo "log_artifact_name=${log_artifact_name}" >> $GITHUB_OUTPUT
+
+    - uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+      with:
+        name: vault-k8s-image
+
+    - name: Load vault-k8s dev image
+      shell: bash
+      run: |
+        docker image load --input vault-k8s-image.docker.tar
+        kind load docker-image --name ${{ inputs.kind-cluster-name }} hashicorp/vault-k8s:0.0.0-dev
+
+    - name: Makefile tests
+      shell: bash
+      working-directory: vault-k8s
+      run: make deploy exercise teardown VAULT_VERSION="${{ inputs.vault-version }}"
```

Comment:
Should this set KIND_CLUSTER_NAME="${{ inputs.kind-cluster-name }}" as well?

Reply:
Not needed, since I reverted some kind setup make targets that I had added in this PR.

Reply:
> May want to set `VERSION=0.0.0-dev` here just so it's explicit for `make deploy`.

Fixed in 892d419d5d62cb3abba18586302c37f120d8f2aa